### ISSUE TITLE
CSS Flexbox Horizontal Overflow

### ISSUE THREAD

**Issue by zme1** [obdurodon/dh_course#84]
Flexbox horizontal overflow
In our debate pages, the webpage would ideally be formatted so that candidate photos appear directly above the container that holds all utterances by any given candidate on immigration. I have the flexbox container set up and overflow scrolls working. The issue, however, lies in that the candidate photos are wrapping vertically inside their window and, as such, force a vertical scroll. I am trying to make it so that the photos appear side-by-side (I have tried my hand with the `flex-wrap: nowrap;` property along with several other manipulations -- to no avail). The code I have so far is below, as well as links to the files on our repository. Thank you!

xhtml: https://github.com/ech45/immigration_station/blob/master/website/xhtml/R...

css: https://github.com/ech45/immigration_station/blob/master/website/css/rep...

Here are the portions of code included in my question:

xhtml:

```
<div class="containerTwo">
            <div class="photos">     <ul class="candidate">
                <li><a href="#BUSH"><img src="../images/bush.jpg" width="225" height="250"/></a></li>
                <li><a href="#CARSON"><img src="../images/carson.jpg" width="225" height="250"/></a> </li>
                <li><a href="#CHRISTIE"><img src="../images/christie.jpg" width="225" height="250"/></a></li>
                <li><a href="#CRUZ"><img src="../images/cruz.jpg" width="225" height="250"/></a></li>
                <li><a href="#FIORINA"><img src="../images/fiorina.jpg" width="225" height="250"/></a></li>
                <li><a href="#HUCKABEE"><img src="../images/huckabee.jpg" width="225" height="250"/></a></li>
                <li><a href="#KASICH"><img src="../images/kasich.jpg" width="225" height="250"/></a></li>
                <li><a href="#PAUL"><img src="../images/paul.jpg" width="225" height="250"/></a></li>
                <li><a href="#RUBIO"><img src="../images/rubio.jpg" width="225" height="250"/></a></li>
                <li><a href="#TRUMP"><img src="../images/trump.jpg" alt="Trump" width="225" height="250"/></a></li>
                <li><a href="#WALKER"><img src="../images/walker.jpg" width="225" height="250"/></a></li>
                
            </ul>
            </div>
```
(the parent div here extends past the debate, which is why it is not closed in this sample)

css:

```
div.containerTwo{
    background: rgba(0, 0, 0, .70);
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    bottom: 0;
    height: calc(100vh + 50px);
}
div.debate{
    border-left: 2px solid yellow;
    padding-left: 1em;
    overflow-y: auto;
    flex: 1 100%;
    background-color: white;
}
div.photos{
    padding-right: 1em;
    flex: 1 200px;
    overflow-x: scroll;
    height: 250px;
}
```

**Comment by zme1** [comment]
Your images are inside <li> elements, which are children of a <ul> parent, and those don't have flex styling. They don't need to, though; in the CSS properties for the <div> element of @class "container", add the following CSS instruction:

`white-space: nowrap;`

The issue is that your <li> elements have white space between them, and the default when displaying inline or inline-block elements horizontally is to break on white space between them if they don't all fit on the same line. If you turn that off at the level of the parent <ul>, you'll get a single, continuous scrolling line of images

**Comment by zme1** [comment]
That worked perfectly to orient the images in a scroll, but now a new issue arises from that. The other item in the flex box container (the debate excerpts) now does not have text wrapping.. Each utterance falls on a single line, which necessitates an irritating horizontal scroll. Is there a property that can be assigned specifically to the debate `div` to overwrite the property inherited from the container??

**Comment by zme1** [comment]
You can assign a unique `@id` (for example, "candidate_photos") to the `ul` that contains the `li` elements that hold the images, and then turn off white-space wrapping only for that specific `ul`:

```
#candidate_photos: {
    white-space: nowrap;
}
```

A CSS selector that begins with a hash mark selects the element for which the rest of the selector text (in this case, "candidate_photos") is the `@id` value.

More generally, it looks as if you use a lot of `@class` attribute values for items that occur only once, such as wrapper `div` elements for a panel on your page. It's probably easier to manage the site if you use `@id` values for items that are unique (since `@id` values must be unique) and `@class` values where you have multiple items that need to be treated the same way. Using a `@class` value just once has the same effect as using an `@id` value, but because `@class` values are meant to define collections of multiple elements, the syntax for managing them is different. For example, if you want to attach an event listener to an item that you've identified with an `@id` value, you can just refer to it directly:

`document.getElementById('stuff').addEventListener('click',do_something,false);`
If you use a @class value, you have to use a numerical index even with one item:

`document.getElementsByClassName('stuff')[0].addEventListener('click',do_something,false);`
That's not a big deal, but it is mildly misleading, and therefore error-prone.

**Comment by zme1** [comment]
Thank you for the help! Turns out that I had accidentally applied to the container element, which affected the orientation of its child debate `div`. I just moved it to the flex item with the photos to fix the issue!
